### PR TITLE
Add github action to create date+sha releases

### PR DESCRIPTION
**Pull request by shellderp**
Publishes snapshots to maven with date/sha added so snapshots aren't overwritten. I want to pull these in in backfila.
Already added this in backfila, check it out:
image

### CODE REVIEW

**Comment by shellderp** [comment]
@swankjesse is this silly?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          npm install -g @misk/cli
+          miskweb ci-build -e
+      - name: Cache Gradle packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Set gradle version
+        run: echo "version=String.valueOf(project.version).replace(\"-SNAPSHOT\", \"\") + '-$(date +%Y%m%d)-$(git rev-parse --short HEAD)-SNAPSHOT'" > hooks.gradle
```
Comment (swankjesse) [inline .github/workflows/build.yml:32]: wanna do hours+minutes so if we do 2 releases in a day the versions always go forward?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Publish packages
+        run: ./gradlew clean uploadArchives
+        env:
+          ORG_GRADLE_PROJECT_SONATYPE_NEXUS_USERNAME: ${{ secrets.SONATYPE_NEXUS_USERNAME }}
```
Comment (swankjesse) [inline .github/workflows/build.yml:36]: might also need our signing key?
`$secrets.ARTIFACT_SIGNING_PRIVATE_KEY` has it

https://github.com/cashapp/certifikit/blob/master/.github/workflows/Release.yml

Comment (swankjesse) [reply .github/workflows/build.yml:36]: (this one is triggered by a tag being pushed)

File: .github/workflows/build.yml (hunk 1 above)
Comment (swankjesse) [reply .github/workflows/build.yml:32]: presumably you use `-SNAPSHOT` so there’s no promotion step?

Comment (shellderp) [reply .github/workflows/build.yml:32]: re: hours, sure why not

re snapshot: I figured it was the less offensive / easier to convince option, but if you think normal release is fine I can try to get that working. There's a promote target in the latest maven plugin

File: .github/workflows/build.yml (hunk 2 above)
Comment (shellderp) [reply .github/workflows/build.yml:36]: somehow I had no problem without it, but I should add it

File: .github/workflows/build.yml (hunk 1 above)
Comment (shellderp) [reply .github/workflows/build.yml]: Added hours+mins and it's not pushing snapshots anymore

**Comment by shellderp** [comment]
@swankjesse PTAL

**Comment by shellderp** [comment]
@swankjesse Update: I think I have it signing, but now it's getting
```
   > Could not PUT 'https://oss.sonatype.org/service/local/staging/deploy/maven2/com/squareup/misk/misk/0.14.0-20200813.0143-2a633e0/misk-0.14.0-20200813.0143-2a633e0.jar'. Received status code 403 from server: Forbidden```

does the cashapp org-level account not have misk publishing perms? had the same issue when trying to upload a snapshot, so I don't think it's a signing problem

File: build.gradle [hunk 3]
```diff
@@ -108,6 +105,17 @@ subprojects { subProject ->
 
   if (!path.startsWith(":samples") && !path.startsWith(":misk-embedded-sample")) {
     apply plugin: 'com.vanniktech.maven.publish'
+
+    // Override signing plugin details that maven-publish will use.
+    signing {
```
Comment (shellderp) [inline build.gradle:110]: this breaks local publishing, so I guess I can put this part in the `hooks.gradle`?

**Comment by chris-ryan-square** [comment]
If this PR is still relevant, then please rebase and get a green CI build.
If the PR is no longer relevant please close.

This PR will be closed at the end of January 2021 otherwise.

**Comment by chris-ryan-square** [comment]
Closing request as per previous announcement.

**Pull request closed without merging**